src/DynamoCoreWpf/NodeViewCustomization/NodeViewCustomizations/FunctionNodeViewCustomization.cs

```diff
@@ -102,7 +103,18 @@ private void EditCustomNodeProperties()
 
                 if (!string.IsNullOrEmpty(ws.FileName))
                 {
-                    ws.Save(ws.FileName, false, dynamoViewModel.EngineController);
+                    string fileContents = File.ReadAllText(ws.FileName);
```

Comment:
instead of reloading the file, we can just find the matching workspace viewModel and call save on it instead I think - that seems faster and safer.

Reply:
since this is a view you have access to the dynamoViewModel where you should be able to get the workspaceViewModel with a matching id for this custom node.

Reply:
I am not sure whether you can do this: `ws.GetStringRepOfWorkspace()`. if needed, this can be changed to `GetStringRepOfWorkspace(engine = null)`.  workspaceViewModel will be listening to `PopulateJSONWorkspace` event...

Reply:
@QilongTang this doesn't work as expected..this gets the model, but not the view.

Reply:
@ramramps Thanks for confirming, I will try grabbing the ViewModel by Id look up

Reply:
@mjkkirschner The solution you suggested does not necessarily working since user could not have that dyf opening currently so it is not loaded in memory FYI: @gregmarr

Reply:
@QilongTang good point - one option is to construct the workspaceViewModel from the workspace instead of looking it up.

Reply:
@mjkkirschner I changed it to using a temp WorkspaceViewModel for serializing and dispose right after.